data/basics/units_test.go

```diff
@@ -37,23 +38,112 @@ func TestSubSaturate(t *testing.T) {
 
 func TestSubSaturate32(t *testing.T) {
 	partitiontest.PartitionTest(t)
+	t.Parallel()
 
-	require.Equal(t, uint32(0), SubSaturate32(0, 1))
-	require.Equal(t, uint32(0), SubSaturate32(1, 2))
-	require.Equal(t, uint32(0), SubSaturate32(1, 1))
-	require.Equal(t, uint32(0), SubSaturate32(1, math.MaxUint32))
-	require.Equal(t, uint32(1), SubSaturate32(2, 1))
-	require.Equal(t, uint32(math.MaxUint32-1), SubSaturate32(math.MaxUint32, 1))
+	require.Equal(t, uint32(0), SubSaturate(uint32(0), uint32(1)))
+	require.Equal(t, uint32(0), SubSaturate(uint32(1), uint32(2)))
+	require.Equal(t, uint32(0), SubSaturate(uint32(1), uint32(1)))
+	require.Equal(t, uint32(0), SubSaturate(uint32(1), math.MaxUint32))
+	require.Equal(t, uint32(1), SubSaturate(uint32(2), uint32(1)))
+	require.Equal(t, uint32(math.MaxUint32-1), SubSaturate(math.MaxUint32, uint32(1)))
 }
 
 func TestAddSaturate32(t *testing.T) {
 	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	require.Equal(t, uint32(1), AddSaturate(uint32(0), uint32(1)))
+	require.Equal(t, uint32(math.MaxUint32-1), AddSaturate(math.MaxUint32-2, uint32(1)))
+	require.Equal(t, uint32(math.MaxUint32), AddSaturate(math.MaxUint32, uint32(0)))
+	require.Equal(t, uint32(math.MaxUint32), AddSaturate(math.MaxUint32-1, uint32(1)))
+	require.Equal(t, uint32(math.MaxUint32), AddSaturate(math.MaxUint32, uint32(2)))
+}
+
+func BenchmarkAddSaturateGenerics(b *testing.B) {
+	b.ResetTimer()
+
+	startVar := uint64(0xdeadbeef)
+	for n := uint64(0); n < uint64(b.N); n++ {
+		temp := AddSaturate(n, startVar)
+		startVar = temp
+	}
+}
+
+// AddSaturateU64Old adds 2 values with saturation on overflow (OLD IMPLEMENTATION)
+func AddSaturateU64Old(a uint64, b uint64) uint64 {
+	res, overflowed := OAdd(a, b)
```

Comment:
Won't this call your new generic implementation, so you're benchmarking against yourself?